### PR TITLE
adds module ec2_vpc_vpn_gateway

### PR DESCRIPTION
**Pull request by klj613**

### CODE REVIEW

**Comment by klj613** [comment]
related to https://github.com/ansible/ansible-modules-extras/pull/1339 and https://github.com/ansible/ansible-modules-extras/pull/1337

**Comment by gregdek** [comment]
Thanks @klj613 for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

**Comment by sivel** [comment]
Here is some initial feedback based on automated scanning of this new module:

```
============================================================================
ec2_vpc_vpn_gateway.py
============================================================================
ERROR: version_added should be 2.1. Currently 2.0
ERROR: sys.exit() call found
```

**Comment by klj613** [comment]
Thanks for the feedback. I will look into this in a day or two.

**Comment by klj613** [comment]
looked at existing modules and opted to use HAS_BOTO logic to remove the sys.exit

**Comment by gregdek** [comment]
Hi @klj613 -- thanks for this new module! 

We've glanced at this module in our weekly module review. New AWS modules are now required to use boto3; therefore, we are putting this module into needs_revision.

When you've made the suggested revisions, please type "ready_for_review" in a comment and we'll put it back into review. Thanks!

**Comment by gregdek** [comment]
(AWS guidelines, for reference: https://github.com/ansible/ansible-modules-extras/blob/devel/cloud/amazon/GUIDELINES.md)

File: cloud/amazon/ec2_vpc_vpn_gateway.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+    required: false
+    default: None
+    aliases: ['ec2_secret_key', 'secret_key']
+  aws_access_key:
+    description:
+      - AWS access key. If not set then the value of the AWS_ACCESS_KEY environment variable is used.
+    required: false
+    default: None
+    aliases: ['ec2_access_key', 'access_key']
+
```
Comment (gundalow) [inline cloud/amazon/ec2_vpc_vpn_gateway.py:59]: If you add this bit of magic you can remove the `region`, `aws_secret_key` and `aws_access_key` bits from help

Compare http://docs.ansible.com/ansible/ec2_ami_module.html
to the documentation in it's yaml

```
extends_documentation_fragment:
     - aws
     - ec2
```

Comment (klj613) [inline cloud/amazon/ec2_vpc_vpn_gateway.py:59]: Thanks, will look into this tonight

**Comment by klj613** [comment]
Will update soon

**Comment by gregdek** [comment]
@klj613 A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes. When you do address the issues, please respond with ready_for_review in your comment, so that we can notify the maintainer.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request closed without merging**